File: Backend/src/main/java/com/pi/credit/Entities/Product.java
package com.pi.credit.Entities;


import jakarta.persistence.*;

import java.util.Date;
import java.util.List;


@Entity
public class Product {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int id;
    private String name;
    private String description;
    @Temporal(TemporalType.DATE)
    private Date Release;
    @ManyToMany
    private List<Category> categories;

    public Product() {
    }

    public int getId() {
        return id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public Date getRelease() {
        return Release;
    }

    public void setRelease(Date release) {
        Release = release;
    }

    public List<Category> getCategories() {
        return categories;
    }

    public void setCategories(List<Category> categories) {
        this.categories = categories;
    }

    @Override
    public String toString() {
        return "Game{" +
                "id=" + id +
                ", name='" + name + '\'' +
                ", description='" + description + '\'' +
                '}';
    }
}

File: Frontend/src/app/models/Product.ts
export interface IProduct {

    id : number;
    name : string;
    description : string;
    Release: Date ;
    categories:any;
    //images:string[];
    //logo:string;
 }
 
 export class Product implements IProduct {
   constructor(
     public id : number,
     public name : string,
     public description : string,
     public Release: Date,
     public categories:any,
     //public logo:string,
    // public images:string[]
   ) { }
 
 
 
 }
 